lib/libzfs/libzfs_mount.c

```diff
@@ -552,8 +552,30 @@ unmount_one(libzfs_handle_t *hdl, const char *mountpoint, int flags)
 	int error;
 
 	error = do_unmount(mountpoint, flags);
+	/* On error, -1 is returned, and errno is set appropriately */
```

Comment:
It feels a little dodgy to depend on `errno` still being set correctly here.  Perhaps we should update `do_unmount` to actually return the `errno` set by the syscall.  This way unrelated changes to `do_unmount()` can't accidentally break this logic.

Reply:
Good point!  I had thought about making `do_unmount` be more like similar functions that return actual error.  I'll make the change.